### PR TITLE
Network ACL lifecycle events

### PR DESCRIPTION
**Pull request by masnax**

### CODE REVIEW

File: lxd/network_acls.go [hunk 1]
```diff
@@ -212,7 +215,7 @@ func networkACLsPost(d *Daemon, r *http.Request) response.Response {
 		return response.BadRequest(err)
 	}
 
-	_, err = acl.LoadByName(d.State(), projectName, req.Name)
+	netACL, err := acl.LoadByName(d.State(), projectName, req.Name)
```
Comment (stgraber) [inline lxd/network_acls.go:218]: This won't work. `netACL` will be `nil` unless it already exists and if it already exists, we return early with an error.
You're then passing `netACL` as `nil` to `NetworkACLCreated` which then panics :)

Comment (masnax) [reply lxd/network_acls.go:218]: Ahhhh one of my major gripes with Go sometimes is the easy-to-miss, yet colossal difference between `err == nil` and `err != nil` :)

Comment (masnax) [reply lxd/network_acls.go:218]: Fixed it by getting the struct again after creating it. Then had to sort out omitting the ACL name in the source by checking if it's the `Created` event in the event handler.

Comment (masnax) [reply lxd/network_acls.go:218]: Actually, I just realized I don't think I do that consistently. I'll make another PR where I add that conditional in all the lifecycle handlers.

**Comment by stgraber** [comment]
@masnax
```
/lxc-ci/build/tmp.WfNqGKvoGx/go/src/github.com/lxc/lxd/lxd/network_acls.go:218:2: ineffectual assignment to netACL
```

**Comment by masnax** [comment]
> @masnax
> 
> ```
> /lxc-ci/build/tmp.WfNqGKvoGx/go/src/github.com/lxc/lxd/lxd/network_acls.go:218:2: ineffectual assignment to netACL
> ```

Oops :)

**Pull request merged**